lib/configs/recommended.json

```diff
@@ -1,7 +1,42 @@
 {
+    "parser": "eslint-plugin-typescript/parser",
+    "parserOptions": {
+        "sourceType": "module"
+    },
+    "plugins": [
+        "typescript"
+    ],
     "rules": {
-        "class-name-casing": "error",
-        "indent": "error",
-        "no-unused-vars": "error"
+        "typescript/adjacent-overload-signatures": "error",
+        "typescript/array-type": "error",
+        "typescript/ban-types": "error",
+        "camelcase": "off",
+        "typescript/camelcase": "error",
+        "typescript/class-name-casing": "error",
+        "typescript/explicit-function-return-type": "warning",
+        "typescript/explicit-member-accessibility": "error",
+        "indent": "off",
+        "typescript/indent": "error",
```

Comment:
`recommended` was designed to only contain possible errors and best practices rules, the `typescript/indent` as a stylistic rule, I'm not sure if it could be included in the recommended.

Reply:
I agree. Although it seems like all `remommended` rulesets out there include stylistic rules as well, probably for the folks who don't use `prettier`. I think the best solution would be to leave these rules in `recommended`, and add `plugin:typescript/prettier` to shut them off. Thoughts?
I also noticed other differences compared to `tslint:recommended`, for example complaining about using `IFoo` as interface name instead of `Foo`. Correct me if I am wrong, but `IFoo` seems like the more popular convention, although on typescript docs site I don't see them used.

Reply:
There’s an `eslint-config-prettier`, which I’m going to add a `typescript` preset to (https://github.com/prettier/eslint-config-prettier/pull/68), which will disable all stylistic rules. People could then extend `["plugin:typescript/recommended", "prettier/typescript"]`.

Reply:
we should not enable stylistic rules as recommended, they are opinionated, some ppl prefer tabs over spaces xd

Reply:
in particular for the second point. There are three possible cases:
1) Don't disable `indent`, don't turn on `typescript/indent`.
    - If they apply the airbnb config, then ours, and we don't disable `indent`, then they'll get a number of incorrect indentation errors due to the incorrect support of TS nodes. This could possibly lead to users opening issues because of the incorrect handling from the base rule.
2) Disable `indent`, don't turn on `typescript/indent`.
    - This is terrible because it means the user will lose their indentation errors, and they mightn't even know we disabled them.
3) Disable `indent`, turn on `typescript/indent`.
    - Stops users using the broken rule, and starts flagging errors. User knows the rule is misconfigured for their code, and adjusts accordingly.

[Looking at some stats](https://ukupat.github.io/tabs-or-spaces/) - of the top 292 starred TS repos; 45% use 4 spaces, 36% use 2 spaces, and 16% use tabs (0.68% uses 3 spaces - wtf).
So with this setting we're likely targeting the most common use case.

Reply:
I don't have any data, but my gut would say that the majority use case is TS only. For those that have a mixed codebase, they're probably advanced enough users that they can apply overrides themselves (not that they need to, `typescript/indent` works fine with plain JS code).

Turning off the base rule is [listed as part of the config in the docs for our indent rule.](https://github.com/bradzacher/eslint-plugin-typescript/blob/master/docs/rules/indent.md#options).

I don't like the idea of only disabling `indent`, whether it be on everything, or just TS(X) files, because unless a user interrogates the recommended config, they won't know we disabled it, and could then fall into using inconsistent indent when they thought their linter was working.

Reply:
what's about vue files? more and more ppl is moving to typescript :)
vue + ts + eslint